apis/core/v1alpha2/core_scope_types.go

```diff
@@ -41,7 +53,22 @@ type HealthScopeSpec struct {
 type HealthScopeStatus struct {
 	runtimev1alpha1.ConditionedStatus `json:",inline"`
 
-	Health string `json:"health"`
+	// ScopeHealthCondition represents health condition summary of the scope
+	ScopeHealthCondition HealthCondition `json:"scopeHealthCondition"`
```

Comment:
they should be different struct as some of the field like componentName don't work here.

Reply:
yes, let me add a specific struct for ScopeHealthCondition,  we can replace
`infoFmtScopeDiagnosis = "total workloads: %d, healthy workloads: %d, unhealthy workloads: %d, unknown workloads: %d"` with the new struct.

Reply:
@wonderflow  any suggestion on the scope health condition struct? e.g. additional info that may be consumed by traits.
```golang
// ScopeHealthCondition represents health condition summary of a scope.
type ScopeHealthCondition struct {
	HealthStatus       HealthStatus `json:"healthStatus"`
	Total              int64        `json:"total,omitempty"`
	HealthyWorkloads   int64        `json:"healthyWorkloads,omitempty"`
	UnhealthyWorkloads int64        `json:"unhealthyWorkloads,omitempty"`
	UnknownWorkloads   int64        `json:"unknownWorkloads,omitempty"`
}

```

Reply:
The struct is good